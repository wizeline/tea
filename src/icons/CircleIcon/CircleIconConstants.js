// @flow

import colors from '../../themes/styles/colors';

const constants = {
  sizes: {
    small: 'small',
    medium: 'medium',
    large: 'large',
  },
  defaultSVGIconColor: colors.white,
  circleDiameters: {
    small: 32,
    medium: 48,
    large: 56,
  },
  tops: {
    small: 8,
    medium: 12,
    large: 12,
  },
  sizeReductions: {
    small: { xsmall: true },
    medium: { small: true },
    large: { medium: true },
  },
};

export default constants;
