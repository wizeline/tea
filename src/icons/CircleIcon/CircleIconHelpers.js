// @flow

const defaultSize = 'small';

const circleDiameters = {
  small: 32,
  medium: 48,
  large: 56,
};

const tops = {
  small: 8,
  medium: 12,
  large: 12,
};

const sizeReductions = {
  small: 'xsmall',
  medium: 'small',
  large: 'medium',
};

const isSizeValid = (size: ?string): boolean =>
  size === undefined || circleDiameters.hasOwnProperty(size);
const parseSize = (size: ?string): string => (!size ? defaultSize : size);
const getDiameter = (parsedSize: string): number => circleDiameters[parsedSize];
const getSquareIconSize = (parsedSize: string): string =>
  sizeReductions[parsedSize];
const getTopOffset = (parsedSize: string): number => tops[parsedSize];

export { isSizeValid, parseSize, getDiameter, getSquareIconSize, getTopOffset };
