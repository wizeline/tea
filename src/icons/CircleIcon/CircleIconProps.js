// @flow

export type Props = {
  name: string,
  color?: string,
  size?: string,
  background?: string,
};

export default {
  defaultProps: {
    color: undefined,
    size: undefined,
    background: '#000000',
  },
};
