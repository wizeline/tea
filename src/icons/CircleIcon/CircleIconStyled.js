// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import styled from 'styled-components';
import type { DiameterProps, BackgroundProps } from './CircleIconTypes';

const getContainerDiameter = ({ diameter }: DiameterProps) => diameter;
const getContainerBackground = (props: BackgroundProps) =>
  props.background || props.theme.backgroundPrimary;

const CircleIconStyled = styled.div`
  display: inline-block;
  height: ${getContainerDiameter}px;
  width: ${getContainerDiameter}px;
  text-align: center;
  background: ${getContainerBackground};
  border-radius: 50%;
`;

export default CircleIconStyled;
