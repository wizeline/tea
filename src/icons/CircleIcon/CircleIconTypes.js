// @flow

export type CircleIconProps = {
  name?: string,
  size?: string,
  color?: string,
  background?: string,
  svgData?: React.Node,
};

export type DiameterProps = {
  size: string,
};

export type BackgroundProps = {
  background: string,
  theme: {
    backgroundPrimary: string,
  },
};
