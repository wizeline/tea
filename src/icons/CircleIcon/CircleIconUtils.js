// @flow

import constants from './CircleIconConstants';
import type { CircleIconProps } from './CircleIconTypes';

const { sizes } = constants;
const parseSize = ({ medium, large }: CircleIconProps): string => {
  if (medium) {
    return sizes.medium;
  }
  if (large) {
    return sizes.large;
  }
  return sizes.small;
};

export default parseSize;
