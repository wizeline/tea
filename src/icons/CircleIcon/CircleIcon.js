// @flow

/* eslint flowtype-errors/enforce-min-coverage: 0 */

import React from 'react';
import SVGIcon from '../SVGIcon';
import type { CircleIconProps } from './CircleIconTypes';
import CircleIconStyled from './CircleIconStyled';
import {
  isSizeValid,
  parseSize,
  getDiameter,
  getSVGIconSize,
  getTopOffset,
} from './CircleIconConstants';

const CircleIcon = ({ name, size, background, color }: CircleIconProps) => {
  if (!isSizeValid(size)) {
    return null;
  }
  const parsedSize: string = parseSize(size);
  return (
    <CircleIconStyled
      diameter={getDiameter(parsedSize)}
      background={background}
    >
      <SVGIcon
        name={name}
        color={color}
        size={getSVGIconSize(parsedSize)}
        top={getTopOffset(parsedSize)}
      />
    </CircleIconStyled>
  );
};

export default CircleIcon;
