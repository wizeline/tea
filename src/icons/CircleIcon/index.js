// @flow
import React from 'react';
import withTheme from '../../utils/withTheme';
import CircleIcon from './CircleIcon';
import ThemeProvider from '../../themes/ThemeProvider';

export default withTheme(CircleIcon);
