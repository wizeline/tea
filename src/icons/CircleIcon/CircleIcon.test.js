import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CircleIcon from './CircleIcon';
import { shallowExpectInvalid } from '../../utils/testHelpers';

describe('CircleIcon', () => {
  it('renders correctly with undefined size attribute', () => {
    const wrapper = shallow(<CircleIcon name="cross" />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('border-radius', '50%');
  });
  it('renders correctly with valid size', () => {
    const wrapper = shallow(<CircleIcon name="cross" size="large" />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('border-radius', '50%');
  });
  it('renders nothing with invalid size', () => {
    shallowExpectInvalid(<CircleIcon name="cross" size="xlarge" />);
  });
});
