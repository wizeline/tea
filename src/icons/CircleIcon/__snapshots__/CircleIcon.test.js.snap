// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleIcon renders correctly when given svgData 1`] = `
<styled.div
  theme={
    Object {
      "backgroundDanger": "#ec5659",
      "backgroundDangerDisabled": "#e2e8ea",
      "backgroundDangerHover": "#e62a2a",
      "backgroundInput": "#ffffff",
      "backgroundPrimary": "#50a5e7",
      "backgroundPrimaryDisabled": "#e2e8ea",
      "backgroundPrimaryHover": "#4793cd",
      "backgroundSecondary": "#ffffff",
      "backgroundSecondaryDisabled": "#ffffff",
      "backgroundSecondaryHover": "#ffffff",
      "textBody": "#3f484b",
      "textDanger": "#ffffff",
      "textDangerDisabled": "#c3cbcf",
      "textDangerHover": "#ffffff",
      "textHeading": "#3f484b",
      "textIcon": "#ffffff",
      "textIconInvert": "#1b2224",
      "textInput": "#3f484b",
      "textInputFocus": "#a2abaf",
      "textInputInvalid": "#e62a2a",
      "textInputPlaceholder": "#a2abaf",
      "textPrimary": "#3f484b",
      "textPrimaryDisabled": "#a2abaf",
      "textPrimaryHover": "#ffffff",
      "textPrimaryInvert": "#ffffff",
      "textSecondary": "#3f484b",
      "textSecondaryDisabled": "#c3cbcf",
      "textSecondaryHover": "#ffffff",
      "textSubtitle": "#a2abaf",
    }
  }
>
  <SVGIcon
    invert={true}
    size="xsmall"
    svgData={<g />}
    theme={
      Object {
        "backgroundDanger": "#ec5659",
        "backgroundDangerDisabled": "#e2e8ea",
        "backgroundDangerHover": "#e62a2a",
        "backgroundInput": "#ffffff",
        "backgroundPrimary": "#50a5e7",
        "backgroundPrimaryDisabled": "#e2e8ea",
        "backgroundPrimaryHover": "#4793cd",
        "backgroundSecondary": "#ffffff",
        "backgroundSecondaryDisabled": "#ffffff",
        "backgroundSecondaryHover": "#ffffff",
        "textBody": "#3f484b",
        "textDanger": "#ffffff",
        "textDangerDisabled": "#c3cbcf",
        "textDangerHover": "#ffffff",
        "textHeading": "#3f484b",
        "textIcon": "#ffffff",
        "textIconInvert": "#1b2224",
        "textInput": "#3f484b",
        "textInputFocus": "#a2abaf",
        "textInputInvalid": "#e62a2a",
        "textInputPlaceholder": "#a2abaf",
        "textPrimary": "#3f484b",
        "textPrimaryDisabled": "#a2abaf",
        "textPrimaryHover": "#ffffff",
        "textPrimaryInvert": "#ffffff",
        "textSecondary": "#3f484b",
        "textSecondaryDisabled": "#c3cbcf",
        "textSecondaryHover": "#ffffff",
        "textSubtitle": "#a2abaf",
      }
    }
    top={8}
  />
</styled.div>
`;

exports[`CircleIcon renders correctly with undefined size attribute 1`] = `
<styled.div
  theme={
    Object {
      "backgroundDanger": "#ec5659",
      "backgroundDangerDisabled": "#e2e8ea",
      "backgroundDangerHover": "#e62a2a",
      "backgroundInput": "#ffffff",
      "backgroundPrimary": "#50a5e7",
      "backgroundPrimaryDisabled": "#e2e8ea",
      "backgroundPrimaryHover": "#4793cd",
      "backgroundSecondary": "#ffffff",
      "backgroundSecondaryDisabled": "#ffffff",
      "backgroundSecondaryHover": "#ffffff",
      "textBody": "#3f484b",
      "textDanger": "#ffffff",
      "textDangerDisabled": "#c3cbcf",
      "textDangerHover": "#ffffff",
      "textHeading": "#3f484b",
      "textIcon": "#ffffff",
      "textIconInvert": "#1b2224",
      "textInput": "#3f484b",
      "textInputFocus": "#a2abaf",
      "textInputInvalid": "#e62a2a",
      "textInputPlaceholder": "#a2abaf",
      "textPrimary": "#3f484b",
      "textPrimaryDisabled": "#a2abaf",
      "textPrimaryHover": "#ffffff",
      "textPrimaryInvert": "#ffffff",
      "textSecondary": "#3f484b",
      "textSecondaryDisabled": "#c3cbcf",
      "textSecondaryHover": "#ffffff",
      "textSubtitle": "#a2abaf",
    }
  }
>
  <SVGIcon
    invert={true}
    size="xsmall"
    svgData={
      <svg
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fillRule="evenodd"
          transform="translate(1 1)"
        >
          <path
            d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
          />
        </g>
      </svg>
    }
    theme={
      Object {
        "backgroundDanger": "#ec5659",
        "backgroundDangerDisabled": "#e2e8ea",
        "backgroundDangerHover": "#e62a2a",
        "backgroundInput": "#ffffff",
        "backgroundPrimary": "#50a5e7",
        "backgroundPrimaryDisabled": "#e2e8ea",
        "backgroundPrimaryHover": "#4793cd",
        "backgroundSecondary": "#ffffff",
        "backgroundSecondaryDisabled": "#ffffff",
        "backgroundSecondaryHover": "#ffffff",
        "textBody": "#3f484b",
        "textDanger": "#ffffff",
        "textDangerDisabled": "#c3cbcf",
        "textDangerHover": "#ffffff",
        "textHeading": "#3f484b",
        "textIcon": "#ffffff",
        "textIconInvert": "#1b2224",
        "textInput": "#3f484b",
        "textInputFocus": "#a2abaf",
        "textInputInvalid": "#e62a2a",
        "textInputPlaceholder": "#a2abaf",
        "textPrimary": "#3f484b",
        "textPrimaryDisabled": "#a2abaf",
        "textPrimaryHover": "#ffffff",
        "textPrimaryInvert": "#ffffff",
        "textSecondary": "#3f484b",
        "textSecondaryDisabled": "#c3cbcf",
        "textSecondaryHover": "#ffffff",
        "textSubtitle": "#a2abaf",
      }
    }
    top={8}
  />
</styled.div>
`;

exports[`CircleIcon renders correctly with valid size 1`] = `
<styled.div
  large={true}
  theme={
    Object {
      "backgroundDanger": "#ec5659",
      "backgroundDangerDisabled": "#e2e8ea",
      "backgroundDangerHover": "#e62a2a",
      "backgroundInput": "#ffffff",
      "backgroundPrimary": "#50a5e7",
      "backgroundPrimaryDisabled": "#e2e8ea",
      "backgroundPrimaryHover": "#4793cd",
      "backgroundSecondary": "#ffffff",
      "backgroundSecondaryDisabled": "#ffffff",
      "backgroundSecondaryHover": "#ffffff",
      "textBody": "#3f484b",
      "textDanger": "#ffffff",
      "textDangerDisabled": "#c3cbcf",
      "textDangerHover": "#ffffff",
      "textHeading": "#3f484b",
      "textIcon": "#ffffff",
      "textIconInvert": "#1b2224",
      "textInput": "#3f484b",
      "textInputFocus": "#a2abaf",
      "textInputInvalid": "#e62a2a",
      "textInputPlaceholder": "#a2abaf",
      "textPrimary": "#3f484b",
      "textPrimaryDisabled": "#a2abaf",
      "textPrimaryHover": "#ffffff",
      "textPrimaryInvert": "#ffffff",
      "textSecondary": "#3f484b",
      "textSecondaryDisabled": "#c3cbcf",
      "textSecondaryHover": "#ffffff",
      "textSubtitle": "#a2abaf",
    }
  }
>
  <SVGIcon
    invert={true}
    size="medium"
    svgData={
      <svg
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fillRule="evenodd"
          transform="translate(1 1)"
        >
          <path
            d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
          />
        </g>
      </svg>
    }
    theme={
      Object {
        "backgroundDanger": "#ec5659",
        "backgroundDangerDisabled": "#e2e8ea",
        "backgroundDangerHover": "#e62a2a",
        "backgroundInput": "#ffffff",
        "backgroundPrimary": "#50a5e7",
        "backgroundPrimaryDisabled": "#e2e8ea",
        "backgroundPrimaryHover": "#4793cd",
        "backgroundSecondary": "#ffffff",
        "backgroundSecondaryDisabled": "#ffffff",
        "backgroundSecondaryHover": "#ffffff",
        "textBody": "#3f484b",
        "textDanger": "#ffffff",
        "textDangerDisabled": "#c3cbcf",
        "textDangerHover": "#ffffff",
        "textHeading": "#3f484b",
        "textIcon": "#ffffff",
        "textIconInvert": "#1b2224",
        "textInput": "#3f484b",
        "textInputFocus": "#a2abaf",
        "textInputInvalid": "#e62a2a",
        "textInputPlaceholder": "#a2abaf",
        "textPrimary": "#3f484b",
        "textPrimaryDisabled": "#a2abaf",
        "textPrimaryHover": "#ffffff",
        "textPrimaryInvert": "#ffffff",
        "textSecondary": "#3f484b",
        "textSecondaryDisabled": "#c3cbcf",
        "textSecondaryHover": "#ffffff",
        "textSubtitle": "#a2abaf",
      }
    }
    top={12}
  />
</styled.div>
`;

exports[`CircleIcon renders correctly with valid size 2`] = `
<styled.div
  medium={true}
  theme={
    Object {
      "backgroundDanger": "#ec5659",
      "backgroundDangerDisabled": "#e2e8ea",
      "backgroundDangerHover": "#e62a2a",
      "backgroundInput": "#ffffff",
      "backgroundPrimary": "#50a5e7",
      "backgroundPrimaryDisabled": "#e2e8ea",
      "backgroundPrimaryHover": "#4793cd",
      "backgroundSecondary": "#ffffff",
      "backgroundSecondaryDisabled": "#ffffff",
      "backgroundSecondaryHover": "#ffffff",
      "textBody": "#3f484b",
      "textDanger": "#ffffff",
      "textDangerDisabled": "#c3cbcf",
      "textDangerHover": "#ffffff",
      "textHeading": "#3f484b",
      "textIcon": "#ffffff",
      "textIconInvert": "#1b2224",
      "textInput": "#3f484b",
      "textInputFocus": "#a2abaf",
      "textInputInvalid": "#e62a2a",
      "textInputPlaceholder": "#a2abaf",
      "textPrimary": "#3f484b",
      "textPrimaryDisabled": "#a2abaf",
      "textPrimaryHover": "#ffffff",
      "textPrimaryInvert": "#ffffff",
      "textSecondary": "#3f484b",
      "textSecondaryDisabled": "#c3cbcf",
      "textSecondaryHover": "#ffffff",
      "textSubtitle": "#a2abaf",
    }
  }
>
  <SVGIcon
    invert={true}
    size="small"
    svgData={
      <svg
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fillRule="evenodd"
          transform="translate(1 1)"
        >
          <path
            d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
          />
        </g>
      </svg>
    }
    theme={
      Object {
        "backgroundDanger": "#ec5659",
        "backgroundDangerDisabled": "#e2e8ea",
        "backgroundDangerHover": "#e62a2a",
        "backgroundInput": "#ffffff",
        "backgroundPrimary": "#50a5e7",
        "backgroundPrimaryDisabled": "#e2e8ea",
        "backgroundPrimaryHover": "#4793cd",
        "backgroundSecondary": "#ffffff",
        "backgroundSecondaryDisabled": "#ffffff",
        "backgroundSecondaryHover": "#ffffff",
        "textBody": "#3f484b",
        "textDanger": "#ffffff",
        "textDangerDisabled": "#c3cbcf",
        "textDangerHover": "#ffffff",
        "textHeading": "#3f484b",
        "textIcon": "#ffffff",
        "textIconInvert": "#1b2224",
        "textInput": "#3f484b",
        "textInputFocus": "#a2abaf",
        "textInputInvalid": "#e62a2a",
        "textInputPlaceholder": "#a2abaf",
        "textPrimary": "#3f484b",
        "textPrimaryDisabled": "#a2abaf",
        "textPrimaryHover": "#ffffff",
        "textPrimaryInvert": "#ffffff",
        "textSecondary": "#3f484b",
        "textSecondaryDisabled": "#c3cbcf",
        "textSecondaryHover": "#ffffff",
        "textSubtitle": "#a2abaf",
      }
    }
    top={12}
  />
</styled.div>
`;

exports[`CircleIcon renders correctly with valid size 3`] = `
<styled.div
  small={true}
  theme={
    Object {
      "backgroundDanger": "#ec5659",
      "backgroundDangerDisabled": "#e2e8ea",
      "backgroundDangerHover": "#e62a2a",
      "backgroundInput": "#ffffff",
      "backgroundPrimary": "#50a5e7",
      "backgroundPrimaryDisabled": "#e2e8ea",
      "backgroundPrimaryHover": "#4793cd",
      "backgroundSecondary": "#ffffff",
      "backgroundSecondaryDisabled": "#ffffff",
      "backgroundSecondaryHover": "#ffffff",
      "textBody": "#3f484b",
      "textDanger": "#ffffff",
      "textDangerDisabled": "#c3cbcf",
      "textDangerHover": "#ffffff",
      "textHeading": "#3f484b",
      "textIcon": "#ffffff",
      "textIconInvert": "#1b2224",
      "textInput": "#3f484b",
      "textInputFocus": "#a2abaf",
      "textInputInvalid": "#e62a2a",
      "textInputPlaceholder": "#a2abaf",
      "textPrimary": "#3f484b",
      "textPrimaryDisabled": "#a2abaf",
      "textPrimaryHover": "#ffffff",
      "textPrimaryInvert": "#ffffff",
      "textSecondary": "#3f484b",
      "textSecondaryDisabled": "#c3cbcf",
      "textSecondaryHover": "#ffffff",
      "textSubtitle": "#a2abaf",
    }
  }
>
  <SVGIcon
    invert={true}
    size="xsmall"
    svgData={
      <svg
        viewBox="0 0 16 16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fillRule="evenodd"
          transform="translate(1 1)"
        >
          <path
            d="M7 5.77L12.77 0 14 1.23 8.23 7 14 12.77 12.77 14 7 8.23 1.23 14 0 12.77 5.77 7 0 1.23 1.23 0 7 5.77z"
          />
        </g>
      </svg>
    }
    theme={
      Object {
        "backgroundDanger": "#ec5659",
        "backgroundDangerDisabled": "#e2e8ea",
        "backgroundDangerHover": "#e62a2a",
        "backgroundInput": "#ffffff",
        "backgroundPrimary": "#50a5e7",
        "backgroundPrimaryDisabled": "#e2e8ea",
        "backgroundPrimaryHover": "#4793cd",
        "backgroundSecondary": "#ffffff",
        "backgroundSecondaryDisabled": "#ffffff",
        "backgroundSecondaryHover": "#ffffff",
        "textBody": "#3f484b",
        "textDanger": "#ffffff",
        "textDangerDisabled": "#c3cbcf",
        "textDangerHover": "#ffffff",
        "textHeading": "#3f484b",
        "textIcon": "#ffffff",
        "textIconInvert": "#1b2224",
        "textInput": "#3f484b",
        "textInputFocus": "#a2abaf",
        "textInputInvalid": "#e62a2a",
        "textInputPlaceholder": "#a2abaf",
        "textPrimary": "#3f484b",
        "textPrimaryDisabled": "#a2abaf",
        "textPrimaryHover": "#ffffff",
        "textPrimaryInvert": "#ffffff",
        "textSecondary": "#3f484b",
        "textSecondaryDisabled": "#c3cbcf",
        "textSecondaryHover": "#ffffff",
        "textSubtitle": "#a2abaf",
      }
    }
    top={8}
  />
</styled.div>
`;

exports[`CircleIcon renders nothing with invalid name 1`] = `""`;
